lib/install/read-shrinkwrap.js

```diff
@@ -20,7 +20,7 @@ function readShrinkwrap (child, next) {
     // Don't read non-root lockfiles
     child.isTop && maybeReadFile('package-lock.json', child),
     child.isTop && maybeReadFile('package.json', child),
-    (shrinkwrap, lockfile, pkgJson) => {
```

Comment:
If you want to actually remove the overhead of calling read, you also have to get rid of the `maybeReadFile('package.json', child),` line above this one.  This is just ignoring the result, not avoiding the read.

Reply:
I was sorry to forget to remove the line `maybeReadFile('package.json', child)`.
I would be more careful to PR. Thanks you for your suggestions. :-D